### PR TITLE
Patchcheck tmp

### PR DESCRIPTION
**Pull request by AA-Turner**
----
:books: Documentation preview :books:: https://cpython-previews--13.org.readthedocs.build/

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -15,8 +19,46 @@ jobs:
...
+      # Adapted from https://github.com/actions/checkout/issues/520#issuecomment-1167205721
+      - name: Fetch merge-base
+        run: >
+          git fetch
+          origin
+          +${{ github.event.pull_request.head.sha }}:origin/${{ github.event.pull_request.head.ref }}
+          --depth=$(( ${{ github.event.pull_request.commits }} + 1 ))
+          --no-tags
+          --prune
+          --no-recurse-submodules
```
Comment (hugovk) [inline .github/workflows/lint.yml:50]: Can we skip these?

We don't have submodules, and we'll this env will be discarded once it's finished, so no need to delete anything via prune first?

```suggestion
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -15,8 +19,46 @@ jobs:
...
+      - name: Fetch base branch
+        run: >
+          git fetch
+          origin
+          +${{ github.event.pull_request.base.sha }}:origin/${{ github.event.pull_request.base.ref }}
+          --depth=5
+          --no-tags
+          --prune
+          --no-recurse-submodules
```
Comment (hugovk) [inline .github/workflows/lint.yml:59]:
```suggestion
```

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -15,8 +19,46 @@ jobs:
...
+          --depth=$(( ${{ github.event.pull_request.commits }} + 1 ))
+          --no-tags
+          --prune
+          --no-recurse-submodules
+      - name: Fetch base branch
+        run: >
+          git fetch
+          origin
+          +${{ github.event.pull_request.base.sha }}:origin/${{ github.event.pull_request.base.ref }}
+          --depth=5
```
Comment (hugovk) [inline .github/workflows/lint.yml:56]: Why 5?

File: .github/workflows/lint.yml (hunk 1 above)
Comment (AA-Turner) [reply .github/workflows/lint.yml:50]: Probably, I used them as the GH checkout workflow includes them.

File: .github/workflows/lint.yml (hunk 3 above)
Comment (AA-Turner) [reply .github/workflows/lint.yml:56]: As the azure job uses a depth of 5 which seemed to work and is simpler than doing the full refspec dance

**Pull request closed without merging**